tools/aux/tpm2_print.c

```diff
@@ -0,0 +1,336 @@
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//**********************************************************************;
+
+#include <stdio.h>
+#include "log.h"
+#include "files.h"
+#include "tpm2_util.h"
+#include "tpm2_alg_util.h"
+#include "tpm2_tool.h"
+
+typedef enum {
+    file_type_unknown = 0,
+    file_type_TPMS_ATTEST
+} file_type_id;
+
+file_type_id file_type = file_type_unknown;
+const char* file_path = "";
+
+#define FILE_2_SHORT_ERROR() LOG_ERR("%s: File too short", (file_path))
+
+static inline void print_yaml_indent(size_t indent_count)
+{
+    while (indent_count-- > 0) {
+        tpm2_tool_output("  ");
+    }
+}
+
+static bool print_clock_info(FILE* fd, size_t indent_count) {
+    bool res;
+    union {
+        UINT8  u8;
+        UINT32 u32;
+        UINT64 u64;
+    } numb;
+
+    res = files_read_64(fd, &numb.u64);
+    if (!res) {
+        goto read_error;
+    }
+    print_yaml_indent(indent_count);
+    tpm2_tool_output("clock: %llu\n", (long long unsigned int)numb.u64);
+
+    res = files_read_32(fd, &numb.u32);
+    if (!res) {
+        goto read_error;
+    }
+    print_yaml_indent(indent_count);
+    tpm2_tool_output("resetCount: %lu\n", (long unsigned int)numb.u32);
+
+    res = files_read_32(fd, &numb.u32);
+    if (!res) {
+        goto read_error;
+    }
+    print_yaml_indent(indent_count);
+    tpm2_tool_output("restartCount: %lu\n", (long unsigned int)numb.u32);
+
+    res = files_read_bytes(fd, &numb.u8, 1);
+    if (!res) {
+        goto read_error;
+    }
+    print_yaml_indent(indent_count);
+    tpm2_tool_output("safe: %u\n", (unsigned int)numb.u8);
+
+    // success
+    return true;
+
+read_error:
+    FILE_2_SHORT_ERROR();
+    return false;
+}
+
+static bool print_TPMS_QUOTE_INFO(FILE* fd, size_t indent_count) {
+    bool res;
+
+    // read TPML_PCR_SELECTION count (UINT32)
+    UINT32 pcr_selection_count;
+    res = files_read_32(fd, &pcr_selection_count);
+    if (!res) {
+        goto read_error;
+    }
+    print_yaml_indent(indent_count);
+    tpm2_tool_output("pcrSelect:\n");
+
+    print_yaml_indent(indent_count + 1);
+    tpm2_tool_output("count: %lu\n", (long unsigned int)pcr_selection_count);
+
+    print_yaml_indent(indent_count + 1);
+    tpm2_tool_output("pcrSelections:\n");
+
+    // read TPML_PCR_SELECTION array (of size count)
+    for(long unsigned int i = 0; i < pcr_selection_count; ++i) {
```

Comment:
This `i` declaration is what should be fixed.

Reply:
Yes keep types consistent, I know their are some sketchy spots in this codebase, but Id like to not add more.

Reply:
Fixed. I guess my compiler defaults to C99 and missed this.

Reply:
I used `long unsigned int` to avoid extra casting on prints below. Longs are guaranteed to be at least 32 bits wide, but I can change it if you prefer.